tests/unit/test_schemas.py

```diff
@@ -236,23 +187,26 @@ def validateClasses(self, factory):
         for cls in protocol.getProtocolClasses():
             instance = factory(cls)
             jsonDict = instance.toJsonDict()
-            self.assertTrue(cls.validate(jsonDict))
+            if not cls.validate(jsonDict):
+                print(cls.__name__)
+                print(avrotools.ValidationTool.getInvalidFields(
+                    cls, jsonDict))
+                self.fail("validation failed")
 
-    @unittest.skipIf(protocol.version.startswith("0.6"), "")
     def testValidateDefaultValues(self):
         self.validateClasses(self.getDefaultInstance)
 
-    @unittest.skipIf(protocol.version.startswith("0.6"), "")
     def testValidateTypicalValues(self):
-        self.validateClasses(self.getTypicalInstance)
+        self.validateClasses(avrotools.CreationTool.getInstance)
 
+    # TODO fix ReadAlignment.alignments decode type ambiguity!
```

Comment:
@jeromekelleher there is a nasty bug here which has to do with some ambiguity in the schemas.  What happens here is that we get nondeterministic failures when `testValidateRandomValues` is enabled when it hits `ReadAlignment` or any type that has `ReadAlignment` embedded in it somewhere.  The reason for this is that `ReadAlignment.alignment`'s signature is:
```
union { null, LinearAlignment, GraphAlignment } alignment = null;
```
When we try and decode a `ReadAlignment` dict into a protocol object, we need to decide what type to interpret the `dict['alignment']` as.  But, unfortunately, the decision is already made for us in `ReadAlignment.getEmbeddedType`, which is obviously wrong when we want to decode the value as a `GraphAlignment`:
```
    @classmethod
    def getEmbeddedType(cls, fieldName):
        embeddedTypes = {
            'alignment': LinearAlignment,  # ugh
            'nextMatePosition': Side,
        }

        return embeddedTypes[fieldName]
```
I guess the solution to this previously was that there was only one non-null type in a union.  But the schemas no longer reflect that convention.

How should we proceed here?